### DISCUSSION TITLE
Behavior of composer run-script -l

### DISCUSSION THREAD

**Discussion by james-tu2** [composer/composer#10705] [Q&A / Support] [answered]
Hello!

I noticed in my recent upgrade to Composer v2.3.3 that the behaviour of `composer run-script -l` has been changed.

It currently throws an argument error:

```
❯ composer run-script -l

  Not enough arguments (missing: "script").

run-script [--timeout TIMEOUT] [--dev] [--no-dev] [-l|--list] [--] <script> [<args>...]
```

whereas in Composer v2.2.11 it will list the scripts without any issues:

```
❯ composer run-script -l
scripts:
  post-autoload-dump
  post-root-package-install
  post-create-project-cmd
  test                       Runs the test script as defined in composer.json.
  post-install-cmd
```

Was this changed intended? I could not find any docs in the release notes about it.

**Comment by mbabker** [comment, marked as answer]
Looks like it changed as part of https://github.com/composer/composer/pull/10476.  The argument should probably be switched back to being optional and checked in the command's `execute()` method.

**Reply by stof** [reply]
indeed. Can you send a PR ?

Note that to satisfy phpstan, the solution is probably to run the check on the `$script` variable instead of making a separate call to `getArgument` for the check.

**Reply by mbabker** [reply]
If nobody beats me to it I can do a PR when I get in from work this evening.

**Comment by stof** [comment]
In the meantime, this has been reported as https://github.com/composer/composer/issues/10707 as well.

**Reply by james-tu2** [reply]
@stof or @mbabker 

Would you like me to create a separate actual issue or will you use #10707 instead?

I created this as a discussion originally since I wasn't sure if the behavioural change was intended.

**Reply by stof** [reply]
use the existing issue

**Comment by james-tu2** [comment]
Thank you everyone.
